lib/model/common/src/drivers/pframe/spec/anchored_id.ts

```diff
@@ -0,0 +1,215 @@
+import canonicalize from 'canonicalize';
+import type { AxisId, PColumnSpec } from './spec';
+import { getAxisId, matchAxisId } from './spec';
+import type { AAxisId, AnchorAxisRef, AnchorAxisRefByIdx, APColumnId, APColumnSelector, PColumnSelector } from './selectors';
+
+//
+// Helper functions
+//
+
+function axisKey(axis: AxisId): string {
+  return canonicalize(getAxisId(axis))!;
+}
+
+function domainKey(key: string, value: string): string {
+  return JSON.stringify([key, value]);
+}
+
+/**
+ * Context for resolving and generating anchored references to columns and axes
+ * Maintains maps of known domain values and axes that can be referenced by anchors
+ */
+export class AnchorCtx {
```

Comment:
Ctx is very overloaded word, could we use another word? It might be `GroupedDomains`, not sure.

Reply:
In this case it seems to be a good word. Can't think of a better name.